enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1379,10 +1384,23 @@ func (p *PebbleCache) userGroupID(ctx context.Context) string {
 	return user.GetGroupID()
 }
 
+func (p *PebbleCache) partitionOverride(ctx context.Context) string {
+	md := metadata.ValueFromIncomingContext(ctx, PartitionOverrideKey)
+	if len(md) > 0 {
+		return md[0]
+	}
+	return ""
+}
+
 func (p *PebbleCache) lookupGroupAndPartitionID(ctx context.Context, remoteInstanceName string) (string, string) {
 	groupID := p.userGroupID(ctx)
+	requestedPartition := p.partitionOverride(ctx)
 	for _, pm := range p.partitionMappings {
-		if pm.GroupID == groupID && strings.HasPrefix(remoteInstanceName, pm.Prefix) {
+		matchesPartitionOverride := requestedPartition != "" &&
+			pm.PartitionID == requestedPartition &&
+			(pm.GroupID == groupID || pm.GroupID == "")
+		matchesPartition := pm.GroupID == groupID && strings.HasPrefix(remoteInstanceName, pm.Prefix)
+		if matchesPartitionOverride || matchesPartition {
 			return groupID, pm.PartitionID
 		}
```

Comment:
this code has auth implications so it seems worth adding/updating a test

Reply:
Good call - helped catch a bug in the logic